### PR TITLE
Net shell refactoring

### PR DESCRIPTION
**Pull request by jukkar**
This PR will refactor network shell to make it nicer to work with. The current net-shell is one big (168kb) monolith file which is difficult to work with. So this PR will refactor it into smaller pieces where each net-shell command is placed into its own file. There are no functional changes in this PR, files are shuffled around and split into smaller pieces. There are lot of commits where each shell command move is in its own commit.

### CODE REVIEW

**Comment by jukkar** [comment]
Note to reviewers and thank you beforehand for all the reviews. Sorry about the amount of commits, but I wanted each net-shell command to one specific commit which then exploded the commit count. Fortunately the individual commits are just snippets taken from main net_shell.c master file. :bow:

**Comment by jukkar** [comment]
CI is failing because some glitch in cellular modem sample.

**Comment by pdgendt** [comment]
Maybe a dumb question, but why are the header files added?

**Comment by jukkar** [comment]
The shell commands were created earlier with `SHELL_STATIC_SUBCMD_SET_CREATE` and I did not want to do too big changes to existing code so used a header file to set the commands. It is possible to use `SHELL_SUBCMD_SET_CREATE` in which case we could put all the commands in their own .c file.
So if you think it makes sense to get rid of those .h file, I can certainly investigate that route too.

**Review by rlubos — COMMENTED** [review]
Nice cleanup, but indeed maybe it'd make sense to investigate whether it'd be possible to get rid of the header files.

**Comment by jukkar** [comment]
New version with the header files removed. It is much cleaner looking now. It took some time to figure out how the shell macros are to be used, the samples did not had similar usage but fortunately shell unit tests had.

**Review by pdgendt — DISMISSED** [review]

**Review by rlubos — DISMISSED** [review]
LGTM

**Review by cfriedt — DISMISSED** [review]
LGTM - a nice follow-up commit at a later time be to conditionally compile some shell functionality based on Kconfig choices.

**Comment by carlescufi** [comment]
@jukkar please rebase

**Review by pdgendt — DISMISSED** [review]
Re-approve, assuming this is only a rebase

**Review by rlubos — DISMISSED** [review]

File: subsys/net/lib/shell/ping.c [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(net_shell);
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <zephyr/random/rand32.h>
```
Comment (pdgendt) [inline subsys/net/lib/shell/ping.c:13]:
```suggestion
#include <zephyr/random/random.h>
```

**Review by pdgendt — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**